networking/src/Ntp/Client.hs

```diff
@@ -298,6 +298,28 @@ spawnNtpClient settings ncStatus = do
     fn :: Last (WithAddrFamily t Socket.Socket) -> IO ()
     fn (Last sock) = Socket.close $ runWithAddrFamily sock
 
+    -- Try to resolve addresses, on failure wait 30s and start again.
+    resolve = do
+        logInfo "resolve DNS"
+        addrs <- catMaybes <$> traverse resolveNtpHost (ntpServers settings)
+        if null addrs
+          then do
+            atomically $ writeTVar ncStatus NtpSyncUnavailable
+            -- either wait 30s or wait for `NtpSyncPending` which might be set
+            -- by a client (e.g. wallet), in which case try to resolve the dns.
+            race_
+              (threadDelay 30000000)
+              (do
+                atomically $ do
+                  s <- readTVar ncStatus
+                  case s of
+                    NtpSyncPending -> return ()
+                    _              -> retry
+                logInfo "NtpSyncPending while waiting"
+              )
+            resolve
```

Comment:
This block changes the NTP semantics right? Now it will keep on retrying, but before it would stop forever if the resolution fails?

Reply:
Yes, that's right.